technologies/job/azure-databricks/default/instanceActions.js

```diff
@@ -0,0 +1,139 @@
+  }
+};
+
+/**
+ * Logic to stop the external job instance.
+ * @param {Object} params
+ * @param {Object} params.job - Contains job data including featuresValues.
+ * @param {Object} params.instance - Contains instance data including the payload returned in the start function.
+ */
+exports.stop = async ({ job, instance }) => {
+  try {
+    console.log('STOP INSTANCE:', instance);
+
+    await axios.post(
+      `https://${job.featuresValues.workspace.url}/api/2.0/jobs/runs/cancel`,
+      {
+        run_id: instance.payload.run_id,
+      },
+      await getHeadersWithAccessTokenForDatabricksResource(job.featuresValues),
+    );
+
+    return Response.success();
+  } catch (error) {
+    return getErrorMessage(error, ERRORS_MESSAGES.FAILED_TO_STOP_JOB_ERROR);
+  }
+};
+
+/**
+ * Logic to retrieve the external job instance status.
+ * @param {Object} params
+ * @param {Object} params.job - Contains job data including featuresValues.
+ * @param {Object} params.instance - Contains instance data including the payload returned in the start function.
+ */
+exports.getStatus = async ({ job, instance }) => {
+  try {
+    console.log('GET STATUS INSTANCE:', instance);
+    const res = await axios.get(
+      `https://${job.featuresValues.workspace.url}/api/2.0/jobs/runs/get?run_id=${instance.payload.run_id}`,
+      await getHeadersWithAccessTokenForDatabricksResource(job.featuresValues),
+    );
+
+    if (res && res.data && res.data.state) {
+      return Response.success(
+        JOB_STATES[res.data.state.life_cycle_state]
+        || JOB_STATES[res.data.state.result_state]
+        || JobStatus.AWAITING
+      );
+    }
+
+    return Response.success(JobStatus.AWAITING);
+  } catch (error) {
+    return getErrorMessage(error, ERRORS_MESSAGES.FAILED_TO_GET_STATUS_ERROR);
+  }
+};
+
+/**
+ * Logic to retrieve the external job instance logs.
+ * @param {Object} params
+ * @param {Object} params.job - Contains job data including featuresValues.
+ * @param {Object} params.instance - Contains instance data including the payload returned in the start function.
+ */
+exports.getLogs = async ({ job, instance }) => {
+  try {
+    console.log('GET LOG INSTANCE:', instance);
+    const res = await axios.get(
+      `https://${job.featuresValues.workspace.url}/api/2.0/jobs/runs/get?run_id=${instance.payload.run_id}`,
+      await getHeadersWithAccessTokenForDatabricksResource(job.featuresValues),
+    );
+
+    if (res && res.data && res.data.cluster_instance && res.data.cluster_instance.cluster_id) {
+      const clusterId = res.data.cluster_instance.cluster_id;
+
+      const resCluster = await axios.get(
+        `https://${job.featuresValues.workspace.url}/api/2.0/clusters/get?cluster_id=${clusterId}`,
+        await getHeadersWithAccessTokenForDatabricksResource(job.featuresValues),
+      );
+
+      const { data : clusterData } = resCluster;
+
+      const logsDestination = clusterData.cluster_log_conf?.dbfs?.destination;
```

Comment:
Are you sure that this syntax works using Nodejs 12.X? I did not try it, but I think the support of optionnal chaining is only available on Nodejs >= 14.